File: GameApplicationJava/code/src/Creatures/AttackType.java
package Creatures;
// define attack types, AtkGrid[2]=Base Attack Structure

class AttackType {

    //blank layout
    public static int[][] Blank() {
        int[][] ATKGrid =
                {{0, 0, 0, 0, 0},
                {0, 0, 0, 0, 0},
                {0, 0, 0, 0, 0},
                {0, 0, 0, 0, 0},
                {0, 0, 0, 0, 0}};
        return ATKGrid;
    }

    //Direct layout, Base = {0,0,1,0,0}
    public static int[][] Direct() {
        int[][] ATKGrid =
                {{1, 0, 0, 0, 0},
                {0, 1, 0, 0, 0},
                {0, 0, 1, 0, 0},
                {0, 0, 0, 1, 0},
                {0, 0, 0, 0, 1}};
        return ATKGrid;
    }

    //BullDozer layout, Base = {0,1,1,1,0}
    public static int[][] BullDozer() {
        int[][] ATKGrid =
                {{1, 1, 0, 0, 0},
                {1, 1, 1, 0, 0},
                {0, 1, 1, 1, 0},
                {0, 0, 1, 1, 1},
                {0, 0, 0, 1, 1}};
        return ATKGrid;
    }

    //FireBreather layout, Base = {1,0,1,0,1}
    public static int[][] FireBreathing() {
        int[][] ATKGrid =
                {{1, 0, 1, 0, 0},
                {0, 1, 0, 1, 0},
                {1, 0, 1, 0, 1},
                {0, 1, 0, 1, 0},
                {0, 0, 1, 0, 1}};
        return ATKGrid;
    }

    //Flanker layout, Base = {1,0,0,0,1}
    public static int[][] Flanker() {
        int[][] ATKGrid =
                {{0, 0, 1, 0, 0},
                {0, 0, 0, 1, 0},
                {1, 0, 0, 0, 1},
                {0, 1, 0, 0, 0},
                {0, 0, 1, 0, 0}};
        return ATKGrid;
    }

    //fog layout, Base = {1,1,1,1,1}
    public static int[][] Fog() {
        int[][] ATKGrid =
                {{1, 1, 1, 0, 0},
                {1, 1, 1, 1, 0},
                {1, 1, 1, 1, 1},
                {0, 1, 1, 1, 1},
                {0, 0, 1, 1, 1}};
        return ATKGrid;
    }


}

File: GameApplicationJavaScript/JavaScript/CreatureTools/AttackType.js
/*
 *2d Arrays which contain the attack structure of each position
 *called only by Creature Maker
 *Alter With Caution: will break game
 */
function AttackType() {
    //if AttackType called without 'new', treat as though it was


    if (!(this instanceof AttackType)) {
        return new AttackType();
    }

    this.BLANK = function () {

        return [
            [0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0]
        ];
    };


    this.BULL_DOZER = function () {
        return [
            [1, 1, 0, 0, 0],
            [1, 1, 1, 0, 0],
            [0, 1, 1, 1, 0],
            [0, 0, 1, 1, 1],
            [0, 0, 0, 1, 1]
        ];
    };

    this.DIRECT = function () {
        return [
            [1, 0, 0, 0, 0],
            [0, 1, 0, 0, 0],
            [0, 0, 1, 0, 0],
            [0, 0, 0, 1, 0],
            [0, 0, 0, 0, 1]
        ];
    };

    this.FIRE_BREATHING = function () {
        return [
            [1, 0, 1, 0, 0],
            [0, 1, 0, 1, 0],
            [1, 0, 1, 0, 1],
            [0, 1, 0, 1, 0],
            [0, 0, 1, 0, 1]
        ];
    };

    this.FLANKER = function () {
        return [
            [0, 0, 1, 0, 0],
            [0, 0, 0, 1, 0],
            [1, 0, 0, 0, 1],
            [0, 1, 0, 0, 0],
            [0, 0, 1, 0, 0]
        ];
    };

    this.FOG = function () {

        return [
            [1, 1, 1, 0, 0],
            [1, 1, 1, 1, 0],
            [1, 1, 1, 1, 1],
            [0, 1, 1, 1, 1],
            [0, 0, 1, 1, 1]
        ];
    };
}